### PR TITLE
Decimal percentile support.

### PR DESCRIPTION
**Pull request by amrishlal**
This PR adds support for specifying decimal percentile (99.999, 0.00001, etc) through the following functions:
```
PERCENTILE(column, percentile)
PERCENTILEEST(column, percentile)
PERCENTILETDIGEST(column, percentile)
PERCENTILEMV(column, percentile)
PERCENTILEESTMV(column, percentile)
PERCENTILETDIGESTMV(column, percentile)
```

Note that there is no change in the external behavior of existing percentile functions, that only take integer percentile values, as described in [documentation](https://docs.pinot.apache.org/users/user-guide-query/supported-aggregations). However, the internal implementation of these functions were changed to support decimal percentile values as well.

Files modified to allow for successfully parsing and validating decimal values in new percentile functions:

> AggregationFunctionFactory.java

Existing implementation of percentile functions were modified to allow for supporting both the existing and the new percentile functions with decimal values:

> PercentileAggregationFunction.java
> PercentileEstAggregationFunction.java
> PercentileEstMVAggregationFunction.java
> PercentileMVAggregationFunction.java
> PercentileTDigestAggregationFunction.java
> PercentileTDigestMVAggregationFunction.java

Files modified to add / fix new unit tests:

> AggregationFunctionFactoryTest.java
> PercentileTDigestQueriesTest.java
> InterSegmentResultTableMultiValueQueriesTest.java
> InterSegmentResultTableSingleValueQueriesTest.java
> PercentileTDigestMVQueriesTest.java
> PercentileTDigestQueriesTest.java

I will update the documentation once this PR is complete.

### CODE REVIEW

**Comment by mayankshriv** [comment]
Thanks for keeping the backward compatibility. We can document and retire the previous version in future release.

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/PercentileTDigestMVAggregationFunction.java [hunk 1]
```diff
@@ -33,19 +33,27 @@ public PercentileTDigestMVAggregationFunction(ExpressionContext expression, int
...
+    return _version == 0 ? AggregationFunctionType.PERCENTILETDIGEST.getName() + (int) _percentile + "MV_" + _expression
+        : AggregationFunctionType.PERCENTILETDIGEST.getName() + _percentile + "MV_" + _expression;
   }
 
   @Override
   public String getResultColumnName() {
-    return AggregationFunctionType.PERCENTILETDIGEST.getName().toLowerCase() + _percentile + "mv(" + _expression + ")";
+    return _version == 0 ? AggregationFunctionType.PERCENTILETDIGEST.getName().toLowerCase() + (int) _percentile + "mv("
+        + _expression + ")"
+        : AggregationFunctionType.PERCENTILETDIGEST.getName().toLowerCase() + "mv(" + _expression + ", " + _percentile
```
Comment (mayankshriv) [inline pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/PercentileTDigestMVAggregationFunction.java:55]: @Jackie-Jiang Thoughts on what should we return as column name for the new way of calling `percentile(column, <double>)`?

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/PercentileTDigestMVAggregationFunction.java:55]: The current way looks good (returning `percentile(column, <double>)`)

**Review by Jackie-Jiang — APPROVED** [review]

**Review by mayankshriv — APPROVED** [review]

**Review by mcvsubbu — COMMENTED** [review]
Can you also add this aggregation in the integration tests please?

**Comment by fx19880617** [comment]
Also, please update the pinot docs for this function :) 
https://docs.pinot.apache.org/users/user-guide-query/supported-aggregations

**Pull request merged**